### ISSUE TITLE
Update `@yamada-ui/alert` accessibility 

### ISSUE THREAD

**Issue by setodeve** [yamada-ui/yamada-ui#1723]
### Description
This issue aims to address accessibility deficiencies in our `alert component` as per W3C guidelines. 
The component currently does not meet the accessibility standards outlined in the W3C documentation. 

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

#### Role
- [ ]  role="alertdialog" is used appropriately.

### Problem Statement/Justification

none

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### References

https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/
https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/

**Comment by Tomoya-Matsubara** [comment]
Can I take this issue?

**Comment by Tomoya-Matsubara** [comment]
Regarding the `alertdialog` role, [W3C First Public Working Draft](https://www.w3.org/TR/wai-aria-1.3/#alertdialog) says:

> The `alertdialog` [role](https://www.w3.org/TR/wai-aria-1.3/#dfn-role) goes on the [node](https://dom.spec.whatwg.org/#concept-node) containing both the alert message and the rest of the dialog.
> ...
> Unlike [alert](https://www.w3.org/TR/wai-aria-1.3/#alert), alertdialog can receive a response from the user.

Currently, the `@yamada-ui/alert` component has only an alert message but doesn't support dialogs. 

The `alertdialog` in question goes on a component like [`AlertDialog` in Chakra UI](https://v2.chakra-ui.com/docs/components/alert-dialog/usage), but apparently there has yet to be no such component in Yamada UI. Therefore, I think this issue can be closed.
